### PR TITLE
Convert text-summarization and translation services to grpc and service.

### PR DESCRIPTION
**Pull request by ferrouswheel**

### CODE REVIEW

File: text-summarization/Dockerfile [hunk 1]
```diff
@@ -17,6 +17,11 @@ RUN apt-get install -y git default-jre && pip3.6 install -r opennmt_requirements
 ADD requirements.txt /service_requirements.txt
 RUN pip3.6 install -r service_requirements.txt
 
+# Install snet daemon
+RUN curl -OL https://github.com/singnet/snet-daemon/releases/download/v0.1.0/snetd-0.1.0.tar.gz && \
+      tar xzvf snetd-0.1.0.tar.gz && \
+      cp snetd-linux-amd64 /usr/local/bin/snetd# Install snet daemon
```
Comment (arturgontijo) [inline text-summarization/Dockerfile]: >cp snetd-linux-amd64 /usr/local/bin/snetd# Install snet daemon

cp snetd-linux-amd64 /usr/local/bin/snetd

Comment (arturgontijo) [reply text-summarization/Dockerfile]: We can use the new daemon release: [snetd-0.1.1.tar.gz](https://github.com/singnet/snet-daemon/releases/download/v0.1.1/snetd-0.1.1.tar.gz)

I think this download must be done inside a folder (like `/tmp`, `/opt/singnet` or even `/opt/singnet`) and then it must be deleted.
In fact I think all these instructions must be in one of these folders.

What do you think about add the `git clone` of the repo inside Dockerfile too?

**Comment by arturgontijo** [comment]
I'm following the instructions from `text-summarization/README.md`:

- Inside the container created by Dockerfile the `git version` is `2.7.4` and it doesn't support `-j8` flag
- `python fetch_models.py` ---> `python3.6 ../fetch_models.py`
- Can't find `fetch_corenlp.sh`
- `python translate.py ...` ---> `cd opennmt-py` and `python3.6 translate.py ...`
  - `-model ../models/gigaword_nocopy_acc_51.33_ppl_12.74_e20.pt \` --> `-model ../models/sum_transformer_model_acc_57.25_ppl_9.22_e16.pt \`, maybe?
Or should we have to download the gigaword from [link](https://s3.amazonaws.com/opennmt-models/gigaword_nocopy_acc_51.33_ppl_12.74_e20.pt)?
  - `-src ~/data/cnndm/test.txt.src` --> `data/test_model2.src`, maybe?

- Then the README ends without a service deploy guide.
What do you think about add this guide?
To help external users that want to publish and test it as a SingularityNET Service.
I think this guide could be added at README or just a link to our `wiki/tutorials` (WIP).

File: .gitignore [hunk 2]
```diff
@@ -114,3 +114,5 @@ venv.bak/
 **/db
 **/*.icloud
 twitter_messages.txt
+
+.vscode/
```
Comment (glautervl) [inline .gitignore]: You might also delete the .vscode path.

**Comment by glautervl** [comment]
For the readme file I think we need add a user guide showing how to call the service.
I think to keep the same pattern for other services that we are coding and add:
- Logs
- Unit and integration tests
- Maintain same folder structure

File: text-summarization/Dockerfile (hunk 1 above)
Comment (ferrouswheel) [reply text-summarization/Dockerfile]: Ok - I will make the first changes you suggest.

Using `git clone` for the repository that the Dockerfile is in is redundant. The repo contents already get uploaded to the docker daemon added as part of the docker context, so we should never `git clone` the original repo without a very good reason.

Comment (arturgontijo) [reply text-summarization/Dockerfile]: My idea was to use `docker build` with your Dockerfile's GitHub path.
But I agree with your way too. So yes, the user must clone the repo just one time.

Some points, following your Dockerfile:
- Keeping your way, I think the `fetch_models.py` will be missing, right?
- What about the `snetd.config.json`?
- To run `buildproto.sh` we need `grpcio-tools==1.14.1` (it's not in `requirements.txt`)

One thing I just saw is that the `snetd` version `0.1.1` changed `--config-path` to just `--config`.

**Comment by ferrouswheel** [comment]
@arturgontijo I'm not using the Dockerfile for development or a tutorial, it is for deployment only. So I'm not too worried about commands being different. I will clarify in the README how to create a virtualenv for the project though, which will provide some isolation of dependencies.

I will confirm the commands in the README work and the models are correct. It might have changed slightly since I first wrote it.

I will also add an example of running and calling the service, but I'll leave service deployment as a link to the wiki once it exists (I think we have guides in too many places and it's going to be annoying to update them all whenever things change!)

@glautervl Those are fair comments. However this PR is primarily to upgrade to grpc.

Note these upgraded NLP services existed before the others, so it's the other services in this repo that redefined the folder structure. I'm happy to agree on a standard, but I'd like some discussion and specific feedback about why the structure I'm using is not acceptable.

I have added issues to add consistent logging and tests to each of these services (#22 #23), since that is outside the scope of this PR.

File: text-summarization/Dockerfile (hunk 1 above)
Comment (ferrouswheel) [reply text-summarization/Dockerfile]: Ah ok. Most build systems will check out the repo as the first step. I haven't seen many people building directly from a remote Dockerfile.

`fetch_models.py` - it is run outside of the docker build, either by the developer or by CircleCI, the downloaded models become part of the docker context. They can be big, so no need to download them for every docker build. Everything in the repo/docker context gets added as part of `ADD . /text-summarization`

The `snetd.config.json` file for the service is hosted on the server. It contains a private key so is not allowed to be in the git repo (we have already had our private keys exposed by people accidentally adding them... so best to never add files with sensitive information into the repo)

I will add `grpcio-tools` to the requirements. I copied the buildproto.sh from another repo that wasn't using virtualenv and so it was calling python3.6 and using my system's grpcio-tools

Thanks for the heads up about the `--config-path` argument changing.

**Comment by ferrouswheel** [comment]
I'm going to merge this as I'm assuming the comments were meant added as part of a code review.

**Pull request merged**